### PR TITLE
Add overwrite flag

### PR DESCRIPTION
**Pull request by akucharska**
**Description**

Changes proposed in this pull request:

- Add overwrite flag that can disable overwriting values when the field is hidden

**Related issue(s)**

### CODE REVIEW

File: docs/extensibility/form-section.md [hunk 1]
```diff
@@ -45,6 +46,16 @@ If you target elements of an array rather than the array itself, you can use the
       name: Item name
     - path: service.url
     - path: service.port
+    - path: service.number
```
Comment (qbalukom) [inline docs/extensibility/form-section.md:49]: you added a mutually exclusive `visibility` and yet you didn't include `overwrite` here. Which should be used.
we could use an example with single `visibility` and no `overwrite` in addition to it as well

File: core-ui/src/components/Extensibility/plugins/VisibilityHandler.js [hunk 2]
```diff
@@ -20,6 +20,10 @@ export function VisibilityHandler({
 
   // rule.visibility won't work for "var", we must use schema.get('visibility')
   const visibilityFormula = schema.get('visibility');
+  const overwrite =
+    typeof schema.get('overwrite') !== 'undefined'
+      ? schema.get('overwrite')
+      : true;
```
Comment (qbalukom) [inline core-ui/src/components/Extensibility/plugins/VisibilityHandler.js:26]: wouldn't `??` be enough here?
```suggestion
  const overwrite = schema.get('overwrite') ?? true;
```

**Review by qbalukom — APPROVED** [review]

File: docs/extensibility/form-section.md [hunk 3]
```diff
@@ -33,6 +33,7 @@ If you target elements of an array rather than the array itself, you can use the
 - **simple** - parameter used to display the simple form. It is `false` by default.
 - **advanced** - parameter used to display the advanced form. It is `true` by default.
 - **visibility** - a [JSONata](jsonata.md) expression controlling the visibility of the element.
+- **overwrite** - parameter used to disable the overwriting (clearing) of hidden fields. Used together with **visibility**, defaults to `true`. **NOTE:** it is recommended to set **overwrite** to `false` when defining fields with the same `path` and different **visibility** conditions.
```
Comment (grego952) [inline docs/extensibility/form-section.md:36]:
```suggestion
- **overwrite** - parameter used to disable the overwriting (clearing) of hidden fields. Used together with **visibility**, defaults to `true`. 
**NOTE:** it is recommended to set **overwrite** to `false` when defining fields with the same `path` and different **visibility** conditions.
```

**Review by grego952 — APPROVED** [review]

**Review by qbalukom — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**